### PR TITLE
Add AuthData to client.Info

### LINKED ISSUE #4418: Add AuthData to client.Info

### PR DESCRIPTION
**Pull request by jpkrohling**
Fixes #4418

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
Skipping changelog, as it should be added as the last step.

File: client/doc_test.go [hunk 1]
```diff
@@ -59,3 +59,32 @@ func Example_processor() {
...
+	_ = client.NewContext(ctx, cl)
+}
+
+type exampleAuthData struct {
+	username string
+}
+
+func (e *exampleAuthData) GetAttribute(string) interface{} {
+	return nil
+}
```
Comment (bogdandrutu) [inline client/doc_test.go:87]: Should this be `if key == "username" return e.username`?

File: client/doc_test.go [hunk 2]
```diff
@@ -59,3 +59,32 @@ func Example_processor() {
...
+func (e *exampleAuthData) GetAttributeNames() []string {
+	return nil
+}
```
Comment (bogdandrutu) [inline client/doc_test.go:90]: Should this `return []string{"username"}`?

**Review by bogdandrutu — APPROVED** [review]

**Comment by jpkrohling** [comment]
@bogdandrutu, @tigrannajaryan, do you mind merging this one?

**Pull request merged**